gui/src/deadliner.rs

```diff
@@ -446,31 +396,111 @@ impl<'a> App for Deadliner<'a> {
+            .selected_text(bg.to_string())
+            .show_ui(ui, |ui| {
+                for option in Background::iter().collect::<Vec<_>>() {
+                    let label = option.to_string();
+                    ui.selectable_value(bg, option, label);
+                }
+            });
+    });
+
+    ui.add_space(PADDING);
+
+    match bg {
+        Background::Solid(color) => {
+            ui.horizontal(|ui| {
+                ui.label("Pick a Color:");
+                ui.color_edit_button_srgb(color);
+            });
+        }
+        Background::FromURL { url, .. } => {
+            ui.horizontal(|ui| {
+                ui.label("Image URL:");
+                ui.add(
+                    egui::TextEdit::singleline(url)
+                        .desired_width(180.)
+                        .hint_text(
+                            RichText::new("https://source.unsplash.com/random")
+                                .color(Color32::from_white_alpha(45)),
+                        ),
+                );
+            });
+        }
+        Background::FromDisk { location, .. } => {
+            ui.horizontal(|ui| {
+                let mut is_valid = true;
+                if ui.button("Open file…").clicked() {
+                    if let Some(path) = rfd::FileDialog::new().pick_file() {
+                        let new_location = path.display().to_string();
+
+                        let file_name = get_file_name_from_path(&new_location);
+                        let file_ext = file_name.split(".").collect::<Vec<&str>>().pop().unwrap();
+                        let supported_file_ext = ["png", "gif", "jpg", "jpeg"];
+
+                        if supported_file_ext.contains(&file_ext) {
+                            *location = new_location;
+                            is_valid = true;
+                        } else {
+                            is_valid = false;
+                        }
+                    }
+                }
+
+                if !is_valid {
+                    ui.colored_label(Color32::from_rgb(255, 48, 48), "Not an Image");
+                } else if !location.is_empty() {
+                    ui.colored_label(
+                        Color32::from_rgba_unmultiplied(254, 216, 67, 200),
+                        get_file_name_from_path(location),
+                    );
+                }
+            });
+        }
+    }
+
+    if let Background::FromDisk { mode, .. } | Background::FromURL { mode, .. } = bg {
+        ui.add_space(PADDING);
+
+        ui.horizontal(|ui| {
+            ui.label("Wallpaper Mode:");
+
+            ComboBox::from_id_source("background_mode")
+                .selected_text(format!("{:?}", mode))
+                .show_ui(ui, |ui| {
+                    for option in WallpaperMode::iter().collect::<Vec<_>>() {
+                        ui.selectable_value(mode, option, format!("{:?}", option));
+                    }
+                });
+        });
+    }
+}
+
```

Comment:
I love that!

yeah, the background mode isn't necessary when it's a solid background cause we generate a solid background image exactly using the monitor resolution so we're pretty sure It fits perfectly!

Reply:
Yeah, that was kind of fun. I don't think I've actually used a match statement with multiple variants binding to the same variable name before. :)